src/sage/modules/free_module_integer.py

```diff
@@ -805,3 +806,73 @@ def CVPP_2V(t, V, voronoi_cell):
             t_new = t_new - CVPP_2V(t_new, V_scaled, ZZ(2 ** (i - 1)) * voronoi_cell)
             i -= 1
         return t - t_new
+
+    def approximate_closest_vector(self, t, delta=None, *args, **kwargs):
+        r"""
+        Compute a vector `w` such that
+
+        .. MATH::
+
+            |t-w|<(\frac{1}{\delta-\frac{1}{4}})^{d/2}|t-u|
+
+        where `u` is the closest lattice point to `t`, `\delta` is the LLL
+        reduction parameter, and `d` is the dimension of the lattice.
+
+        This will check whether the basis is already `\delta`-LLL-reduced
+        and otherwise it will run LLL to make sure that it is. For more
+        information about ``delta`` see :meth:`LLL`.
+
+        INPUT:
+
+        - ``t`` -- the target vector to compute a close vector to
+
+        - ``delta`` -- (default: ``0.99``) the LLL reduction parameter
+
+        - ``*args`` -- passed through to :meth:`LLL`
+
+        - ``**kwds`` -- passed through to :meth:`LLL`
+
+        OUTPUT:
+
+        The vector `w` described above.
+
+        EXAMPLES::
+
+            sage: from sage.modules.free_module_integer import IntegerLattice
+            sage: L = IntegerLattice([[1, 0], [0, 1]])
+            sage: L.approximate_closest_vector((-6, 5/3))
+            (-6, 2)
+
+        The quality of the approximation depends on ``delta``::
+
+            sage: from sage.modules.free_module_integer import IntegerLattice
+            sage: L = IntegerLattice([[101, 0, 0, 0], [0, 101, 0, 0],
+            ....:                     [0, 0, 101, 0], [-28, 39, 45, 1]], lll_reduce=False)
+            sage: t = vector([1337]*4)
+            sage: L.approximate_closest_vector(t, delta=0.26)
+            (1348, 1340, 1383, 1337)
+            sage: L.approximate_closest_vector(t, delta=0.99)
+            (1326, 1349, 1339, 1345)
+            sage: L.closest_vector(t)
+            (1326, 1349, 1339, 1345)
+
+        ALGORITHM:
+
+        Uses the algorithm from [Bab86]_.
+        """
+        if delta is None:
+            delta = ZZ(99)/ZZ(100)
+
+        # bound checks on delta are performed in is_LLL_reduced
+        if not self._reduced_basis.is_LLL_reduced(delta=delta):
+            self.LLL(*args, delta=delta, **kwargs)
+
+        B = self._reduced_basis
+        G = B.gram_schmidt()[0]
+        t = vector(t)
+
+        b = t
+        for i in reversed(range(G.nrows())):
+            b -= B[i] * ((b * G[i]) / (G[i] * G[i])).round("even")
+        return (t - b).change_ring(ZZ)
```

Comment:
```suggestion
        return (t - b).change_ring(ZZ)
        
```
Sorry I missed this. There should be an extra line. Otherwise it is good. Thanks :)